internal/indexcoord/node_manager.go

```diff
@@ -34,16 +34,18 @@ import (
 
 // NodeManager is used by IndexCoord to manage the client of IndexNode.
 type NodeManager struct {
-	nodeClients map[UniqueID]types.IndexNode
-	pq          *PriorityQueue
-	lock        sync.RWMutex
-	ctx         context.Context
+	nodeClients   map[UniqueID]types.IndexNode
+	stoppingNodes map[UniqueID]struct{}
```

Comment:
when do you remove stopping nodes?

Reply:
In the `RemoveNode` method, which is called when the node has stopped.
![image](https://user-images.githubusercontent.com/21985684/207863486-699959c3-21cb-4f8b-abf0-45b678d8d657.png)